bundles/org.openhab.binding.konnected/src/main/java/org/openhab/binding/konnected/internal/handler/KonnectedHandler.java

```diff
@@ -526,6 +527,7 @@ private void sendSetSwitchState(String thingId, String payloadString) throws Kon
     }
 
     private int getOnState(Channel channel) {
-        return ((Number) channel.getConfiguration().get(CHANNEL_ONVALUE)).intValue();
+        Number configuredOnValue = (Number) (channel.getConfiguration().get(CHANNEL_ONVALUE));
```

Comment:
If you would like to consolidate default values for this configuration, you could consider providing a DTO class and use `getConfiguration().as(ClassName.class)` here. Example with some default values:
https://github.com/openhab/openhab-addons/blob/5a11f1420ed201a1dcca24e82251917c2d9d72f0/bundles/org.openhab.binding.boschindego/src/main/java/org/openhab/binding/boschindego/internal/config/BoschIndegoConfiguration.java#L27-L28

And usage:
https://github.com/openhab/openhab-addons/blob/5a11f1420ed201a1dcca24e82251917c2d9d72f0/bundles/org.openhab.binding.boschindego/src/main/java/org/openhab/binding/boschindego/internal/handler/BoschIndegoHandler.java#L113-L115

Reply:
Thanks, I looked at it again and it does seem to clean up the code by using a DTC class for the channel config.